jaypeasee/rancid-tomatillos
# Feature: Fetching Movies

# Pull Request Template

#### What’s this PR do?
This PR does three different API calls, replacing the static `movieData`. The first API call fetches all of the movies for the gallery view. The second and third API calls use `Promise.all` to render the picked movie, all of it's details and it's trailer. the currentMovie state property gets updated as one object with more properties now too.

#### Where should the reviewer start?
1. The `componentDidMount()` method in `App.js` and the `getAllMovies()` method in `apiCalls.js`.
1. The `showChosenMovie()` method in `App.js` and the `getMovieByID()` and `getMovieTrailerByID()` methods in `apiCalls.js`.
1. All of `MoviewView.js`.


#### How should this be manually tested?
* Run `npm start` to load the page and see the movies appear.
* Click a movie in the gallery and see the individual movie review appear with the trailer at the bottom.

#### Any background context you want to provide? 
This completes Iteration 3 of the project (minus testing).

#### What are the relevant tickets? 
* https://app.asana.com/0/1199396273595458/1199403385786658
* https://app.asana.com/0/1199396273595458/1199521921795261
* https://app.asana.com/0/1199396273595458/1199511063946465

#### Screenshots (if appropriate): 

<img width="1426" alt="Screen Shot 2020-12-04 at 7 15 24 PM" src="https://user-images.githubusercontent.com/59029768/101230844-12cdb500-3665-11eb-9c26-7290c3f9f99c.png">
